geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySenderEventProcessor.java

```diff
@@ -221,21 +222,17 @@ public RegionQueue getQueue() {
   }
 
   /**
-   * Increment the batch id. This method is not synchronized because this dispatcher is the caller
+   * Increment the batch id.
    */
   public void incrementBatchId() {
-    // If _batchId + 1 == maximum, then roll over
-    if (this.batchId + 1 == Integer.MAX_VALUE) {
-      this.batchId = -1;
-    }
-    this.batchId++;
+    batchId.getAndUpdate(x -> x == Integer.MAX_VALUE ? -1 : x + 1);
```

Comment:
I'm not sure whats happening to the batchIds, but I see these messages on the receiver every time a batch is sent:
```
[warn 2021/05/10 17:08:12.558 PDT ny-1 <ServerConnection on port 5444 Thread 1> tid=0x66] Received process batch request 3 out of order. The id of the last batch processed was 1. This batch request will be processed, but some messages may have been lost.

[warn 2021/05/10 17:08:12.564 PDT ny-1 <ServerConnection on port 5444 Thread 1> tid=0x66] Batch IDs are out of order. Setting latestBatchId to: 3. It was: 1

[warn 2021/05/10 17:09:20.638 PDT ny-1 <ServerConnection on port 5444 Thread 1> tid=0x66] Received process batch request 5 out of order. The id of the last batch processed was 3. This batch request will be processed, but some messages may have been lost.

[warn 2021/05/10 17:09:20.644 PDT ny-1 <ServerConnection on port 5444 Thread 1> tid=0x66] Batch IDs are out of order. Setting latestBatchId to: 5. It was: 3
```